### PR TITLE
Improve telemetry collection in Promscale

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
This PR improves the telemetry collection in Promscale, both for Metrics & Traces. This is an initial Pr, that contains a subset of metrics planned to be supported. A different PR will increase the telemetry count to match the complete list of requirements.

Note: Not ready for review.

The telemetry table now looks like:

Note: `￾￾` are null chars, that are added as padding by name syscall.

```postgresql
test=# select * from _timescaledb_catalog.metadata order by key;
                    key                    |                               value                               | include_in_telemetry 
-------------------------------------------+-------------------------------------------------------------------+----------------------
 install_timestamp                         | 2021-11-16 16:43:35.012198+05:30                                  | t
 promscale_arch                            | amd64                                                             | t
 promscale_commit_hash                     |                                                                   | t
 promscale_default_chunk_interval          | 08:00:00                                                          | t
 promscale_default_metric_retention        | 90 days                                                           | t
 promscale_downsampling_count              | 0                                                                 | t
 promscale_exec_platform                   |                                                                   | t
 promscale_exemplar_metrics_count          | 0                                                                 | t
 promscale_ha_cluster_count                | 0                                                                 | t
 promscale_ingested_samples_total          | 0                                                                 | t
 promscale_instance_count                  | 2                                                                 | t
 promscale_metric_metadata_count           | 0                                                                 | t
 promscale_metrics_count                   | 0                                                                 | t
 promscale_multi_tenancy_tenant_count      | 0                                                                 | t
 promscale_node_count                      | 0                                                                 | t
 promscale_os                              | linux                                                             | t
 promscale_os_domain_name                  | (none)￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾ | t
 promscale_os_machine                      | x86_64￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾ | t
 promscale_os_node_name                    | harkishen￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾ | t
 promscale_os_release                      | 5.11.0-40-generic￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾ | t
 promscale_os_sys_name                     | Linux￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾￾ | t
 promscale_os_version                      | #44~20.04.2-Ubuntu SMP Tue Oct 26 18:07:44 UTC 2021￾￾￾￾￾￾￾￾￾￾￾￾￾￾ | t
 promscale_promql_executed_queries         | 0                                                                 | t
 promscale_promql_exeuction_time_p50       | 0                                                                 | t
 promscale_promql_exeuction_time_p90       | 0                                                                 | t
 promscale_promql_exeuction_time_p95       | 0                                                                 | t
 promscale_promql_exeuction_time_p99       | 0                                                                 | t
 promscale_promql_failed_queries           | 0                                                                 | t
 promscale_promql_timed_out_queries        | 0                                                                 | t
 promscale_series_count_approx             | 0                                                                 | t
 promscale_spans_bytes_total               | 65536                                                             | t
 promscale_spans_count                     | 0                                                                 | t
 promscale_trace_query_exeuction_time_p50  | 0                                                                 | t
 promscale_trace_query_exeuction_time_p90  | 0                                                                 | t
 promscale_trace_query_exeuction_time_p95  | 0                                                                 | t
 promscale_trace_query_exeuction_time_p99  | 0                                                                 | t
 promscale_traces_count                    | 0                                                                 | t
 promscale_traces_dependency_queries_total | 0                                                                 | t
 promscale_traces_queries_total            | 0                                                                 | t
 promscale_version                         | 0.7.0-beta.1.dev.2                                                | t
 uuid                                      | 83ef45e4-b0f4-41a9-a52b-1cdd2fa89b4c                              | t
(41 rows)
```

The promscale_instance_information looks like:

```postgresql
                 uuid                 |           last_updated           | telemetry_metrics_ingested_samples | telemetry_metrics_queries_executed | telemetry_metrics_queries_timed_out | telemetry_metrics_queries_failed | telemetry_traces_queries_executed | telemetry_traces_dependency_queries_executed | deletable 
--------------------------------------+----------------------------------+------------------------------------+------------------------------------+-------------------------------------+----------------------------------+-----------------------------------+----------------------------------------------+-----------
 00000000-0000-0000-0000-000000000000 | 2021-11-16 16:43:35.303567+05:30 |                                  0 |                                  0 |                                   0 |                                0 |                                 0 |                                            0 | f
 c939f575-ba5a-42ca-9606-693ec577b22e | 2021-11-16 16:44:01.72887+05:30  |                                  0 |                                  0 |                                   0 |                                0 |                                 0 |                                            0 | t
(2 rows)
```

### CODE REVIEW

**Comment by paulfantom** [comment]
What is `uuid` related to?

File: pkg/version/version.go [hunk 1]
```diff
@@ -38,8 +38,8 @@ var (
 	// It is customary to bump the version by incrementing the numeral after
 	// the `dev` tag. The SQL migration script name must correspond to the /new/ version.
 
-	Promscale                           = "0.7.0-beta.1.dev.1"
-	PrevReleaseVersion                  = "0.7.0-beta.1"
+	Promscale                           = "0.7.0-beta.1.dev.2"
```
Comment (VineethReddy02) [inline pkg/version/version.go:41]: Any specific reason behind the version change?

Comment (Harkishen-Singh) [reply pkg/version/version.go:41]: because we want to run migration files for telemetry (`sql/versions/dev/0.7.0-beta.1.dev/2-telemetry`).

**Comment by Harkishen-Singh** [comment]
If anyone wants to review the PR, this PR is broken down into series of smaller PRs, starting with #895.

**Pull request closed without merging**